### PR TITLE
Withdraw to asset endpoint

### PR DESCRIPTION
**Pull request by Buckram123**
This PR aims to add `withdraw_to_asset` endpoint for the carrot-app, to allow withdrawing liquidity in provided asset
TBD:
- [x] Handle slippage

### CODE REVIEW

File: contracts/carrot-app/src/handlers/execute.rs [hunk 1]
```diff
@@ -253,8 +259,110 @@ fn withdraw(
     Ok(app_response)
 }
 
-/// Auto-compound the position with earned fees and incentives.
+fn withdraw_to_asset(
+    deps: DepsMut,
+    env: Env,
+    info: MessageInfo,
+    amount: Uint128,
```
Comment (CyberHoward) [inline contracts/carrot-app/src/handlers/execute.rs:266]: I thought we decided that this `amount` will stay the LP amount. Vlad has already implemented the logic that was needed for this change so we should stick with that. 

This function should only swap one of the withdrawn assets into the other.

File: contracts/carrot-app/src/handlers/execute.rs [hunk 2]
```diff
@@ -253,8 +259,110 @@ fn withdraw(
...
+    to_asset: AssetEntry,
+    max_spread: Option<Decimal>,
+    app: App,
+) -> Result<cosmwasm_std::Response, AppError> {
+    // Only the authorized addresses (admin ?) can withdraw
+    app.admin.assert_admin(deps.as_ref(), &info.sender)?;
```
[the remark this answers is not in the document]
Comment (Buckram123) [reply contracts/carrot-app/src/handlers/execute.rs:272]: Should we make methods callable only by top level owner of account?

**Review by CyberHoward — CHANGES_REQUESTED** [review]
Update logic to keep LP token as amount. 

Add a clear doc-comment in the msg.rs to the `amount` field, explaining that it expects LP token amount.

File: contracts/carrot-app/src/handlers/execute.rs (hunk 1 above)
Comment (Buckram123) [reply contracts/carrot-app/src/handlers/execute.rs:266]: > I thought we decided that this `amount` will stay the LP amount. Vlad has already implemented the logic that was needed for this change so we should stick with that.
> This function should only swap one of the withdrawn assets into the other.

I know that Vlad implemented logic to calculate amount of tokens from amount of liquidity, it's still liquidity amount for `withdraw` method. But I was pretty sure the idea of new method `withdraw_to_asset` was to withdraw to asset amount.

Also want to clear up if `to_asset` have to be only one of the two assets from the pool or could be any?

Comment (CyberHoward) [reply contracts/carrot-app/src/handlers/execute.rs:266]: `withdraw_to_asset` should take the lp token `amount` and use that amount to withdraw. The tokens received from that withdraw should then be used to swap to the `to_asset` denom. 

We could limit it to just allow it to the specific token but we might as well accept any token and rely on ANS to resolve it!

Comment (Buckram123) [reply contracts/carrot-app/src/handlers/execute.rs:266]: Gotcha, thanks!

**Review by CyberHoward — APPROVED** [review]

**Pull request merged**